pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authentication/AuthenticationProvider.java

```diff
@@ -61,7 +87,9 @@ public interface AuthenticationProvider extends Closeable {
      * @return the "role" string for the authenticated connection, if the authentication was successful
      * @throws AuthenticationException
      *             if the credentials are not valid
+     * @deprecated use and implement {@link AuthenticationProvider#authenticateAsync(AuthenticationDataSource)} instead.
      */
+    @Deprecated
```

Comment:
Could you explain more about it? IMO, it should not be deprecated. I tried to implement an `AuthenticationProvider` that only overrides the `authenticationAsync` method but it never went here. Instead, the `authenticate` method was still called. There is no test to show it should be deprecated.